File: cakk-common/src/main/java/com/cakk/common/exception/CakkException.java
package com.cakk.common.exception;

import lombok.Getter;

import com.cakk.common.enums.ReturnCode;

@Getter
public class CakkException extends RuntimeException {

	private final ReturnCode returnCode;
	private final String code;
	private final String message;

	public CakkException(ReturnCode returnCode) {
		this.returnCode = returnCode;
		this.code = returnCode.getCode();
		this.message = returnCode.getMessage();
	}
}

File: cakk-common/src/main/kotlin/com/cakk/common/exception/CakkException.kt
package com.cakk.common.exception

import com.cakk.common.enums.ReturnCode

class CakkException(
	private val returnCode: ReturnCode
) : RuntimeException() {

    private val code
		get() = returnCode.code
    override val message: String
		get() = returnCode.message

	fun getReturnCode(): ReturnCode {
		return returnCode
	}
}
